sdccc/build.gradle.kts

```diff
@@ -0,0 +1,191 @@
+    api(libs.org.bouncycastle.bcpkix.jdk15on)
+    api(libs.com.lmax.disruptor)
+    api(libs.jakarta.inject.jakarta.inject.api)
+    api(libs.org.jetbrains.kotlin.kotlin.stdlib)
+    api(libs.org.jetbrains.kotlin.kotlin.reflect)
+    api(libs.com.lemonappdev.konsist)
+    api(libs.com.google.code.gson.gson)
+    testImplementation(libs.org.mockito.mockito.core)
+    testImplementation(libs.org.mockito.kotlin.mockito.kotlin)
+    testImplementation(project(":biceps-model"))
+    testImplementation(project(":dpws-model"))
+    testImplementation(libs.com.tngtech.archunit.archunit.junit5)
+    testImplementation(libs.org.junit.jupiter.junit.jupiter.params)
+    testImplementation(libs.org.jetbrains.kotlin.kotlin.test.junit5)
+}
+
+description = "sdccc"
+
+val testsJar by tasks.registering(Jar::class) {
+    archiveClassifier.set("tests")
+    from(sourceSets["test"].output)
+}
+
+(publishing.publications["maven"] as MavenPublication).artifact(testsJar)
+
+tasks.withType<JavaCompile> {
+    options.release.set(17)
+}
+
+tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
+    kotlinOptions {
+        jvmTarget = "17"
+    }
+}
+
+tasks.check {
+    dependsOn(detektTask)
+    dependsOn("spotbugsMain")
+    dependsOn("spotbugsTest")
+    dependsOn("checkstyleMain", "checkstyleTest")
+    dependsOn("downloadAndUnpackJre")
+}
+
+tasks.withType<com.github.spotbugs.snom.SpotBugsTask> {
+    excludeFilter.set(file("$projectDir/../dev_config/filter.xml"))
+    reports.create("xml") { required = true }
+}
+
+checkstyle {
+    configFile = file("../checkstyle/checkstyle.xml")
+}
+
+spotless {
+    java {
+        palantirJavaFormat()
+        target("src/**/*.java")
+    }
+}
+
+val jreDirectoryName = "jdk-17.0.5+8-jre"
+val jreBasePath = "jre"
+val jreFullPath = "${jreBasePath}/${jreDirectoryName}"
+val jreDownloadUrlPrefix = "https://github.com/adoptium/temurin17-binaries/releases/download/"
+val jreDownloadFileName = "OpenJDK17U-jre_x64_windows_hotspot_17.0.5_8.zip"
+val jreDownloadUrlSuffix = "jdk-17.0.5%2B8/${jreDownloadFileName}"
+val jreDownloadUrl = "${jreDownloadUrlPrefix}${jreDownloadUrlSuffix}"
+
+tasks.register<de.undercouch.gradle.tasks.download.Download>("downloadJre") {
+    src(jreDownloadUrl)
+    dest(file("${layout.buildDirectory.get().asFile}/${jreBasePath}/${jreDownloadFileName}"))
+    overwrite(false)
+    onlyIfModified(true)
+}
+
+tasks.register<Copy>("unpackJre") {
+    dependsOn("downloadJre")
+    from(zipTree(file("${layout.buildDirectory.get().asFile}/${jreBasePath}/${jreDownloadFileName}")))
+    into("${layout.buildDirectory.get().asFile}/${jreBasePath}")
+}
+
+tasks.register("downloadAndUnpackJre") {
+    dependsOn("unpackJre")
+}
+
+tasks.register<Copy>("copyRuntimeLibs") {
+    from(configurations.runtimeClasspath) {
+        exclude { it.isDirectory }
+    }
+    into("${layout.buildDirectory.get().asFile}/lib")
+}
+
+val projectName = "SDCcc-gradle"
+
+launch4j {
```

Comment:
Consider providing something we can reuse in other places instead of hardcoding it here - convention plugin? :)

Reply:
Other places where?

Reply:
sdccc-internal basically, having to repeat almost the same configuration there isn't great